addons/purchase/models/purchase.py

```diff
@@ -1187,60 +1190,61 @@ def onchange_product_id_warning(self):
-        if price_unit and seller and self.order_id.currency_id and seller.currency_id != self.order_id.currency_id:
-            price_unit = seller.currency_id._convert(
-                price_unit, self.order_id.currency_id, self.order_id.company_id, self.date_order or fields.Date.today())
-
-        if seller and self.product_uom and seller.product_uom != self.product_uom:
-            price_unit = seller.product_uom._compute_price(price_unit, self.product_uom)
-
-        self.price_unit = price_unit
-
+    @api.depends('product_qty', 'product_uom')
+    def _compute_price_unit_and_date_planned(self):
+        # record product names to avoid resetting custom descriptions
         default_names = []
         vendors = self.product_id._prepare_sellers({})
         for vendor in vendors:
             product_ctx = {'seller_id': vendor.id, 'lang': get_lang(self.env, self.partner_id.lang).code}
             default_names.append(self._get_product_purchase_description(self.product_id.with_context(product_ctx)))
 
-        if (self.name in default_names or not self.name):
-            product_ctx = {'seller_id': seller.id, 'lang': get_lang(self.env, self.partner_id.lang).code}
-            self.name = self._get_product_purchase_description(self.product_id.with_context(product_ctx))
+        for line in self:
+            if not line.product_id:
+                continue
+            params = {'order_id': line.order_id}
+            seller = line.product_id._select_seller(
+                partner_id=line.partner_id,
+                quantity=line.product_qty,
+                date=line.order_id.date_order and line.order_id.date_order.date(),
+                uom_id=line.product_uom,
+                params=params)
+
+            if seller or not line.date_planned:
+                line.date_planned = line._get_date_planned(seller).strftime(DEFAULT_SERVER_DATETIME_FORMAT)
+
+            # If not seller, use the standard price. It needs a proper currency conversion.
+            if not seller:
+                po_line_uom = line.product_uom or line.product_id.uom_po_id
+                price_unit = line.env['account.tax']._fix_tax_included_price_company(
+                    line.product_id.uom_id._compute_price(line.product_id.standard_price, po_line_uom),
+                    line.product_id.supplier_taxes_id,
+                    line.taxes_id,
+                    line.company_id,
+                )
+                if price_unit and line.order_id.currency_id and line.order_id.company_id.currency_id != line.order_id.currency_id:
+                    price_unit = line.order_id.company_id.currency_id._convert(
+                        price_unit,
+                        line.order_id.currency_id,
+                        line.order_id.company_id,
+                        line.date_order or fields.Date.today(),
```

Comment:
`date_order` is required and equals to `fields.Datetime.now` by default
```suggestion
                        line.date_order,
```